var/spack/repos/builtin/packages/paradiseo/package.py

```diff
@@ -0,0 +1,71 @@
+from spack import *
+import sys
+
+class Paradiseo(Package):
+    """A C++ white-box object-oriented framework dedicated to the reusable design of metaheuristics."""
+    homepage = "http://paradiseo.gforge.inria.fr/"
+
+    # Installing from the development version is a better option at this 
+    # point than using the very old supplied packages
+    version('head', git='https://gforge.inria.fr/git/paradiseo/paradiseo.git')
+    # This is a version that the package formula author has tested successfully.
+    # However, the clone is very large (~1Gb git history). The history in the 
+    # head version has been trimmed significantly.
+    version('dev-safe', git='https://gforge.inria.fr/git/paradiseo/paradiseo.git',
+             commit='dbb8fbe9a786efd4d1c26408ac1883442e7643a6')
+
+    variant('mpi',      default=True,  description='Compile with parallel and distributed metaheuristics module')
+    variant('smp',      default=True,  description='Compile with symmetric multi-processing module ')
+    variant('edo',      default=True,  description='Compile with (Experimental) EDO module')
+    #variant('tests',    default=False, description='Compile with build tests')
+    #variant('doc',      default=False, description='Compile with documentation')
+    variant('debug',    default=False, description='Builds a debug version of the libraries')
+    
+    # Required dependencies
+    depends_on ("cmake")
+
+    # Optional dependencies
+    depends_on ("mpi", when="+mpi")
+    depends_on ("doxygen", when='+doc')
+    
+    # Patches
+    patch('enable_eoserial.patch')
+    patch('fix_osx_detection.patch')
+    patch('fix_tests.patch')
+    patch('fix_tutorials.patch')
+
+    def install(self, spec, prefix):
+        options = []
+        options.extend(std_cmake_args)
+
+        dsuf = 'dylib' if sys.platform == 'darwin' else 'so'
+        options.extend([
+            '-DCMAKE_BUILD_TYPE:STRING=%s' % ('DEBUG' if '+debug' in spec else 'RELEASE'),
```

Comment:
Usually, these are `Debug` and `Release` rather than all-cap, but if this is what the project expects, that's fine.